src/components/engagement/handlers/set-initial-end-date.handler.ts

```diff
@@ -81,22 +94,22 @@ export class SetInitialEndDate implements IEventHandler<SubscribedEvent> {
   private async updateEngagementInitialEndDate(
     engagement: Engagement,
     initialEndDate: CalendarDate | null | undefined,
-    session: Session
+    changeset?: ID
   ) {
     const updateInput = {
-      id: engagement.id,
       initialEndDate: initialEndDate || null,
     };
-    // TODO: Refactor to call repository directly instead of engagementService methods
     if (engagement.__typename === 'LanguageEngagement') {
-      return await this.engagementService.updateLanguageEngagement(
+      await this.engagementRepo.updateLanguageProperties(
```

Comment:
Instead of using the update service method (prevent UpdateEngagementEvent triggering), I used the repo method.

Reply:
Fine with me, less work to do 👍